### PR TITLE
Fix layer group sub selection when toggling a layer back on

### LINKED ISSUE #1078: [Bug]: Layer group selection - UI and loaded layer are out of sync when toggling a layer back on
### What happened?

Activate a layer within a group besides the default. For example, choose Rainfall anomaly, 1-month
Activate another WMS / raster layer. For example, NDVI
Now reactivate Rainfall anomaly. The UI for the layer selection shows 1-month, but the 10-day layer is loaded instead

### Which country / deployment are you running?

Zimbabwe, main but this happens for any country

### Add a screenshot (if relevant)

Screenshot 2024-01-24 at 22 54 27

### PR DESCRIPTION
**Pull request by echaidemenos**
### Description

This fixes #1078 .

## How to test the feature:

- select a layer with group selection and change it to eg. 1 month
- switch between layers check that the loaded layer and the label are consistent 

## Checklist - did you ...

Test your changes with

- [ ] `REACT_APP_COUNTRY=rbd yarn start`
- [ ] `REACT_APP_COUNTRY=cambodia yarn start`
- [ ] `REACT_APP_COUNTRY=mozambique yarn start`
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: frontend/src/components/MapView/LeftPanel/layersPanel/MenuItem/MenuSwitch/SwitchItem/index.tsx [hunk 1]
```diff
@@ -86,7 +86,8 @@ const SwitchItem = memo(
       return selectedActiveLayer.length > 0
         ? selectedActiveLayer[0].id
         : layer.id;
-    }, [layer.id, selectedActiveLayer]);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [layer.id]);
```
[the remark this answers is not in the document]
Comment (ericboucher) [reply frontend/src/components/MapView/LeftPanel/layersPanel/MenuItem/MenuSwitch/SwitchItem/index.tsx:90]: @wadhwamatic what do you prefer?

**Review by ericboucher — APPROVED** [review]

File: frontend/src/components/MapView/LeftPanel/layersPanel/MenuItem/MenuSwitch/SwitchItem/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (wadhwamatic) [reply frontend/src/components/MapView/LeftPanel/layersPanel/MenuItem/MenuSwitch/SwitchItem/index.tsx:90]: As mentioned in https://github.com/WFP-VAM/prism-app/pull/1131, let's keep the opacity persistent as you've implemented it.

[the remark this answers is not in the document]
Comment (ericboucher) [reply frontend/src/components/MapView/LeftPanel/layersPanel/MenuItem/MenuSwitch/SwitchItem/index.tsx:90]: I believe this is not about the opacity but about the selected layer, eg 3m, 9m... @wadhwamatic

[the remark this answers is not in the document]
Comment (wadhwamatic) [reply frontend/src/components/MapView/LeftPanel/layersPanel/MenuItem/MenuSwitch/SwitchItem/index.tsx:90]: sorry, I misread the context here! Yes, remembering the selected layer within the group is good

**Pull request merged**